### PR TITLE
Add xpack.eql.enabled feature flag, disabled by default. Enabled for gradle run task and integration tests.

### PR DESCRIPTION
**Pull request by aleksmaus**
Related to https://github.com/elastic/elasticsearch/issues/49581

### CODE REVIEW

File: distribution/build.gradle [hunk 1]
```diff
@@ -422,6 +422,7 @@ testClusters {
     if (System.getProperty('run.distribution', 'default') == 'default') {
       String licenseType = System.getProperty("run.license_type", "basic")
       if (licenseType == 'trial') {
+        setting 'xpack.eql.enabled', 'true'
```
Comment (imotov) [inline distribution/build.gradle]: We enable it here only for trial, but we run integration tests under basic. That doesn't seem to be consistent. I would just put next to `pack.sql.enabled` until we figure out what to do with it.

**Review by imotov — COMMENTED** [review]
Left a small comment about enabling it by default in run. Not sure if we should enable it in run at all. @colings86 what do you think? Otherwise looks good to me.

File: distribution/build.gradle (hunk 1 above)
Comment (aleksmaus) [reply distribution/build.gradle]: Updated to enable consistently with sql plugin for run task, seems logical.

**Comment by aleksmaus** [comment]
Thought it would be convenient for development to have this enabled for gradle run task, especially since this is already done for sql plugin.

**Comment by colings86** [comment]
on mobile so maybe missing something...

We should definitely have it enabled for the tests in the EQL module but not sure we need it enabled for all tests. We don’t do that for security AFAIK

@jasontedor what are you doing for Autoscaling? Just enable it in the Autoscaling tests?

**Comment by colings86** [comment]
Oh I see we actually do it for security sorry. However I think we should match Autoscaling while it’s in development

**Comment by jasontedor** [comment]
@colings86 We are only going to enable it where it's required in autoscaling tests. I don't think these should be enabled in `run` tasks (if someone wants them, `-Dtests.es.xpack.eql.enabled=true` would do it).

**Review by imotov — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

**Pull request merged**